crates/shim/src/error.rs

```diff
@@ -84,6 +84,9 @@ pub enum Error {
     #[error("Failed to send exit event: {0}")]
     Send(#[from] std::sync::mpsc::SendError<ExitEvent>),
 
+    #[error("Failed to read file: {0}")]
+    Io(#[from] std::io::Error),
```

Comment:
Will this type of error always be `failed to read a file`?

Reply:
I don't really like the requirement to map errors and use macros everywhere, like
`map_err(io_error!`. I feel like [`anyhow.Context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#example) would be a better fit here.

Reply:
However, anyhow error is not easy to downcast if you need check the exact type of error

Reply:
Where exactly we need to downcast?

Reply:
> Where exactly we need to downcast?

When you want to ignore some specific error.